### PR TITLE
[UX][Bugfix] Improve consistency between global and recipient notification preferences

### PR DESCRIPTION
**Pull request by jkdf2**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Nexus 6P running 6.0.1 (CM 13.0)
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

Consistency is a [widely accepted usability standard](https://dl.acm.org/citation.cfm?id=67910). These changes are intended to improve the consistency in the Preferences-related activities.

* ~~Use SwitchPreferenceCompat globally. The current implementation mixes CheckBoxPreferences and SwitchPreferenceCompats for boolean preferences.~~

* ~~Toggle LEDs on/off using a switch. This holds consistency with the current method of toggling of Notifications on/off via a switch, with customization options following the switch. In the current approach, one must unintuitively choose Color -> None.~~

* Bugfix: Until now, changing a recipient preference from "Default" to another ringtone and back will have a radio icon indicating that the previous other ringtone is currently selected, never going back to "Default"

* Recipient-specific default notification and vibration settings now say "Signal default ($ACTUAL_SETTING)" instead of just "Default". This reflects how the global notification setting also says "Default ringtone ($SYSTEM_DEFAULT)".

* Clean-up: Global notification "summary" field is not used, rather it is set to the current ringtone by the RingtoneSummaryListener. Removed this unused field.

If there is any particular change that you disagree with, please notify me and I will remove it from the PR.

----------

## Global settings before

Global before

## Global settings after

Global after

## Contact settings before

Contacts before

## Contact settings after

Contacts after

### CODE REVIEW

**Comment by 2-4601** [comment]
> - Use SwitchPreferenceCompat globally. The current implementation mixes CheckBoxPreferences and SwitchPreferenceCompats for boolean preferences.

https://github.com/WhisperSystems/Signal-Android/issues/5861#issuecomment-260552015
> the convention we use is that top-level booleans are sliders and booleans which are sub-prefs (dependent on a top-level) are checkboxes

**Comment by jkdf2** [comment]
Thanks @2-4601, reverted that one to keep with those conventions.

**Comment by jkdf2** [comment]
New afters:

After, LED checked

After, LED unchecked

**Comment by moxie0** [comment]
Please remove the additional LED preference and the string "Signal" from "Signal default" Thanks!

**Comment by jkdf2** [comment]
@moxie0 Done, as well as a few improvements and a `BooleanListPreference` to allow `pref_led_blink` to depend on `pref_led_color`. See the new After screenshots:

Global notifications disabled

LED notifications disabled

All notifications enabled

**Comment by jeremymasters** [comment]
This is probably nit-picking, but do the circle-checkmark things have to be red?  That makes it look kind of "emergency-ish".  Could they be Signal blue?

**Comment by RiseT** [comment]
@jeremymasters I guess this is just some Android theme he's using. Note that the displayed font is non-standard as well.

**Comment by jkdf2** [comment]
Hey @jeremymasters,

@RiseT is correct, the checkbox is red just due to my system theme. With AOSP stock theme, they are stylized blue.

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 1]
```diff
@@ -491,6 +496,27 @@ public static String getNotificationRingtone(Context context) {
     return getStringPreference(context, RINGTONE_PREF, Settings.System.DEFAULT_NOTIFICATION_URI.toString());
   }
 
+  @Nullable
+  public static String getNotificationRingtoneFriendlyName(Context context) {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: this method does not belong here, this class is a simple preference accessor

**Comment by jkdf2** [comment]
@moxie0 Okay, I had just thought it might be useful to get a ringtone name as a preference getter, but I moved it as a private instance method in `RecipientPreferenceActivity`.

**Comment by jkdf2** [comment]
Merged `master` and fixed merge conflict with #4957.

I have been running this since January with great success 😺 

Anyone else up for giving a 👍 ?

@moxie0 Any feedback?

**Comment by moxie0** [comment]
Sorry but there's a lot going on here and I don't totally understand what's happening at a glance.  It seems like maybe you could break this up into smaller PRs that would be easier to merge?  I can't tell if this is supposed to do a single thing or several things.

**Comment by jkdf2** [comment]
@moxie0 At your request, I have subdivided this PR into the following: #6482, #6481, #6480

Thanks!

**Pull request closed without merging**